ivadomed/main.py

```diff
@@ -362,185 +415,401 @@ def run_command(context, n_gif=0, thr_increment=None, resume_training=False, no_
-    elif command == "test":
-        imed_utils.display_selected_transfoms(transformation_dict, dataset_type=["testing"])
 
-    # Check if multiple raters
-    check_multiple_raters(command == "train", loader_params)
+def execute_training_bids(bids_df, context, loader_params, model_params, n_gif, path_output,
+                          resume_training, train_lst, transform_train_params, transform_valid_params, valid_lst):
 
-    if command == 'train':
-        # Get Validation dataset
-        ds_valid = get_dataset(bids_df, loader_params, valid_lst, transform_valid_params, cuda_available, device,
-                               'validation')
+    cuda_available, device = imed_utils.define_device(context[ConfigKW.GPU_IDS][0])
+
+    # Get Validation BIDS dataset
+    ds_valid = get_dataset(bids_df, loader_params, valid_lst, transform_valid_params, cuda_available, device,
+                           'validation')
+    # Get Training BIDS dataset
+    ds_train = get_dataset(bids_df, loader_params, train_lst, transform_train_params, cuda_available, device,
+                           'training')
+    metric_fns = imed_metrics.get_metric_fns(ds_train.task)
+    # FiLM not supported, not gonna normalize data
+    train_onehotencoder = None
+    # Model directory
+    create_path_model(context, model_params, ds_train, path_output, train_onehotencoder)
+    save_config_file(context, path_output)
+    ###########################
+    # Run the training and validation to gather the metrics
+    # The main training process
+    ###########################
+    best_training_dice, \
+    best_training_loss, \
+    best_validation_dice, \
+    best_validation_loss = imed_training.train(
+        model_params=model_params,
+        dataset_train=ds_train,
+        dataset_val=ds_valid,
+        training_params=context[ConfigKW.TRAINING_PARAMETERS],
+        wandb_params=context.get(ConfigKW.WANDB),
+        path_output=path_output,
+        device=device,
+        cuda_available=cuda_available,
+        metric_fns=metric_fns,
+        n_gif=n_gif,
+        resume_training=resume_training,
+        debugging=context[ConfigKW.DEBUGGING]
+    )
+    return best_training_dice, best_training_loss, best_validation_dice, best_validation_loss, ds_valid
+
+
+def execute_multi_path_training(context: dict,
+                                model_params: dict,
+                                n_gif: int,
+                                path_output: str,
+                                resume_training: bool,
+                                transform_train_params: dict,
+                                transform_valid_params: dict):
+
+    cuda_available, device = imed_utils.define_device(context.get(ConfigKW.GPU_IDS)[0])
+
+    # Probably need to invoke the aggregator around here.
+    # Parse the DataSets keys.
+    ####
+    # BIDS Data Set Section
+    ####
+
+    #
+    from ivadomed.loader.all_dataset_group import AllDatasetGroups
```

Comment:
I'm confused here.
My understanding was that the new "execute_multi_path_training" that we are reviewing now is for the "FILES" path only (as in the config in `example_loader_v2_configs.py`).

Here the code is under " # BIDS Data Set Section" and the "# File Data Set Section" on line 706 is empty.
I also assume that the "# RegEx Data Set Section" section on line 710 is empty as well and that the code from line 713 apllies for all 3 sections (BIDS, FILES and REGEX).

So just to be sure, the goal here is to review for the new FILES path only (not the new BIDS one)?

Reply:
Yes, that is a great point. This entire section of the code is only used when doing multi_path training (see upstream screenshot) (as oppose to the original V1 path)

![image](https://user-images.githubusercontent.com/6720068/200344079-4b7d2eb2-0a64-43c4-9261-bb496e6c612c.png)

The comment section is a bit out dated now. I will update it. 

The multi-path code instantiate object types based on JSON data object type FileDataset or BIDSDataset (V2) or RegexDataset type. So far, we only implemented FIleDatset type but we plan to support long term merging of all other data type as well. 

I will correct the comment here.